bitswap.go

```diff
@@ -258,11 +260,13 @@ func (bs *Bitswap) receiveBlockFrom(blk blocks.Block, from peer.ID) error {
 
 	bs.engine.AddBlock(blk)
 
-	select {
-	case bs.newBlocks <- blk.Cid():
-		// send block off to be reprovided
-	case <-bs.process.Closing():
-		return bs.process.Close()
+	if ProvideEnabled {
```

Comment:
Any reason to make this a _global_ flag? That's generally a last resort for things that should only be changed when debugging (or things that absolutely need to apply to the entire program like "force private networks").

cc @michaelavila